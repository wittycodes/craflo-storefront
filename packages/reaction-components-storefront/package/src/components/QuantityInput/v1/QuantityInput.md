### Overview

The `QuantityInput` component currently is built with [material-ui](https://material-ui.com/) components.
 * [TextField](https://material-ui.com/demos/text-fields/)
 * [InputAdornment](https://material-ui.com/api/input-adornment/)
 * [ButtonBase](https://material-ui.com/api/button-base/)
 * [withStyles](https://material-ui.com/customization/overrides/)

#### Usage

```jsx
<QuantityInput value={3} onChange={(value => console.log("QuantityInput changed", value))} />
```

