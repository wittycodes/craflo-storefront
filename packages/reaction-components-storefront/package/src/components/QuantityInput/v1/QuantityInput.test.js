import React from "react";
import renderer from "react-test-renderer";
import QuantityInput from "./QuantityInput";

test("basic snapshot", () => {
  const component = renderer.create(<QuantityInput />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
