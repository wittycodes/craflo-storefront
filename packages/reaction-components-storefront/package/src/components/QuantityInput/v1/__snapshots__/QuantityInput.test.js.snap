// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiFormControl-root-4"
>
  <div
    className="MuiInputBase-root-21 MuiInput-root-8 QuantityInput-quantityContainer-2 MuiInputBase-formControl-22 MuiInput-formControl-9 MuiInputBase-adornedStart-25 MuiInputBase-adornedEnd-26"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root-38"
    >
      <button
        className="MuiButtonBase-root-42 QuantityInput-incrementButton-1"
        color="default"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
        variant="outlined"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-45"
          focusable="false"
          role="presentation"
          style={
            Object {
              "fontSize": "14px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
    </div>
    <input
      aria-invalid={false}
      className="MuiInputBase-input-31 MuiInput-input-16 QuantityInput-quantityInput-3 MuiInputBase-inputAdornedStart-36 MuiInputBase-inputAdornedEnd-37"
      disabled={false}
      id="addToCartQuantityInput"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value={0}
    />
    <div
      className="MuiInputAdornment-root-38"
    >
      <button
        className="MuiButtonBase-root-42 QuantityInput-incrementButton-1"
        color="default"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
        variant="outlined"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-45"
          focusable="false"
          role="presentation"
          style={
            Object {
              "fontSize": "14px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
